tests/integ/scala/test_dataframe_reader_suite.py

```diff
@@ -219,6 +219,36 @@ def test_read_csv(session, mode):
     assert "Numeric value 'one' is not recognized" in ex_info.value.message
 
 
+@pytest.mark.parametrize("mode", ["select", "copy"])
+def test_read_format_csv(session, mode):
+    reader = get_reader(session, mode)
+    test_file_on_stage = f"@{tmp_stage_name1}/{test_file_csv}"
+    df1 = reader.format("csv").schema(user_schema).load(test_file_on_stage)
+    res = df1.collect()
+    res.sort(key=lambda x: x[0])
+
+    assert len(res) == 2
+    assert len(res[0]) == 3
+    assert res == [Row(1, "one", 1.2), Row(2, "two", 2.2)]
+
+    with pytest.raises(SnowparkDataframeReaderException):
+        session.read.format("csv").load(test_file_on_stage)
```

Comment:
could you also add a `match=...` to the `pytest.raises` so that we match a regex of the error message that is generated